pandas/_libs/index.pyx

```diff
@@ -255,7 +255,7 @@ cdef class IndexEngine:
 
     def get_indexer(self, values):
         self._ensure_mapping_populated()
-        return self.mapping.lookup(values)
+        return self.mapping.lookup(values).astype('intp')
```

Comment:
hmm we don't want to astype, rather change the underlying

Reply:
gotcha, changed the return values in HashTable below (failure unrelated)